src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3485,6 +3485,75 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
             return Operation.CreateOperationNone(semanticModel: null, operation.Syntax, operation.ConstantValue, childrenBuilder.ToImmutableAndFree(), operation.IsImplicit);
         }
 
+        public override IOperation VisitInterpolatedString(IInterpolatedStringOperation operation, int? captureIdForResult)
+        {
+            var partsBuilder = ArrayBuilder<IInterpolatedStringContentOperation>.GetInstance(operation.Parts.Length);
+            foreach (IInterpolatedStringContentOperation element in operation.Parts)
```

Comment:
>foreach [](start = 12, length = 7)

Why is this done in two passes instead of one?

Reply:
We need to visit all interpolations, pushing the constituents parts on evalStack, before we can generate rewritten IInterpolationOperation nodes.

---
In reply to: [185855615](https://github.com/dotnet/roslyn/pull/26545#discussion_r185855615) [](ancestors = 185855615)